File: backend/src/main/java/t/lab/guide/service/AuthService.java
package t.lab.guide.service;

import t.lab.guide.dto.auth.*;

public interface AuthService {
    RegistrationResponse registrationUser (RegistrationRequest request);
    AuthResponse authenticationUser (AuthRequest request);
    void logoutUser (LogoutRequest request);
    TokenPairResponse changePassword (ChangePasswordRequest request);
    TokenPairResponse refreshToken (RefreshRequest request);

    //TODO когда будут entity(model), добавить сервисные методы
}

File: backend/src/main/kotlin/t/lab/guide/service/AuthService.kt
package t.lab.guide.service

import t.lab.guide.dto.auth.AuthResponse
import t.lab.guide.dto.auth.RegistrationResponse
import t.lab.guide.dto.auth.TokenPairResponse
import t.lab.guide.dto.auth.command.AuthCommand
import t.lab.guide.dto.auth.command.ChangePasswordCommand
import t.lab.guide.dto.auth.command.LogoutCommand
import t.lab.guide.dto.auth.command.RefreshCommand
import t.lab.guide.dto.auth.command.RegistrationCommand

interface AuthService {
    fun registerUser(request: RegistrationCommand): RegistrationResponse

    fun authenticateUser(request: AuthCommand): AuthResponse

    fun logoutUser(request: LogoutCommand)

    fun changePassword(request: ChangePasswordCommand): TokenPairResponse

    fun refreshToken(request: RefreshCommand): TokenPairResponse
// TODO когда будут entity(model), добавить сервисные методы
}
